example/commitpr/main.go

```diff
@@ -45,43 +46,47 @@ var (
 	prSubject     = flag.String("pr-title", "", "Title of the pull request. If not specified, no pull request will be created.")
 	prDescription = flag.String("pr-text", "", "Text to put in the description of the pull request.")
-	sourceFiles   = flag.String("files", "", "Comma-separated list of files to commit and their location. The local file is separated by his target location by a semi-colon. If the file should be in the same location with the same name, you can just put the file name and omit the repetition. Example: README.md,main.go:github/examples/commitpr/main.go`")
-	authorName    = flag.String("author-name", "", "Name of the author of the commit.")
-	authorEmail   = flag.String("author-email", "", "Email of the author of the commit.")
+	sourceFiles   = flag.String("files", "", `Comma-separated list of files to commit and their location.
+The local file is separated by its target location by a semi-colon.
+If the file should be in the same location with the same name, you can just put the file name and omit the repetition.
+Example: README.md,main.go:github/examples/commitpr/main.go`)
+	authorName  = flag.String("author-name", "", "Name of the author of the commit.")
+	authorEmail = flag.String("author-email", "", "Email of the author of the commit.")
 )
 
 var client *github.Client
 var ctx = context.Background()
 
-// getRef returns the commit branch reference object if it exists or creates it from the base branch before returning it
+// getRef returns the commit branch reference object if it exists or creates it
+// from the base branch before returning it.
 func getRef() (ref *github.Reference, err error) {
 	if ref, _, err = client.Git.GetRef(ctx, *sourceOwner, *sourceRepo, "refs/heads/"+*commitBranch); err == nil {
-		return
+		return ref, nil
 	}
-	// We consider that an error means the branch has not been found and needs to be created
+	// We consider that an error means the branch has not been found and needs to
+	// be created.
 	if *commitBranch != *baseBranch {
 		if *baseBranch == "" {
-			log.Fatal("The `-base-branch` should not be set to an empty string when the branch specified by `-commit-branch` does not exists")
+			return nil, errors.New("The `-base-branch` should not be set to an empty string when the branch specified by `-commit-branch` does not exists")
 		}
 		var baseRef *github.Reference
 		if baseRef, _, err = client.Git.GetRef(ctx, *sourceOwner, *sourceRepo, "refs/heads/"+*baseBranch); err != nil {
-			return
+			return nil, err
 		}
 		newRef := &github.Reference{Ref: github.String("refs/heads/" + *commitBranch), Object: &github.GitObject{SHA: baseRef.Object.SHA}}
-		if ref, _, err = client.Git.CreateRef(ctx, *sourceOwner, *sourceRepo, newRef); err != nil {
-			return
-		}
+		ref, _, err = client.Git.CreateRef(ctx, *sourceOwner, *sourceRepo, newRef)
+		return ref, err
 	}
-	return
+	return nil, errors.New("The commit branch does not exist but `-base-branch` is the same as `-commit-branch`")
```

Comment:
Next nit...  we prefer to make the "normal" flow be the "main" flow and return errors as quickly as we know them, and finally to reduce the overall indentation.  So this might be better written:

```go
	if *commitBranch == *baseBranch {
		return nil, errors.New("The commit branch does not exist but `-base-branch` is the same as `-commit-branch`")
	}
...
	return ref, err
```